channel.go

```diff
@@ -1,14 +1,33 @@
 package thingfulx
 
-import (
-	"time"
-)
+import "time"
 
 // Channel represents an individual data channel being published by a Thing out
 // there in the world somewhere.
 type Channel struct {
-	ID         string    `json:"id"`
-	Value      string    `json:"value"`
+	ID           string            `json:"id"`
+	Metadata     []ChannelMetadata `json:"metadata"`
+	Observations []Observation     `json:"observations"`
+	Data         Data              `json:"data"`
+}
+
+// ChannelMetadata is an object which contains any metadata required to
+// semantically descibe a measurement collected by a Thing.
+type ChannelMetadata struct {
+	Prop string `json:"property"`
+	Val  string `json:"value"`
+}
```

Comment:
@mdevilliers - I don't think we can do that because a property might be something like `m3-lite:hasDomainOfInterest` while the value is: `thingful:Environment`. See https://gist.github.com/smulube/3a1706e7a70c647a7dc5dbda569c127b for some other examples.